From b71eef563e8832299f441d725c4a6916ea03e221 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Thu, 16 Feb 2023 17:01:16 +0100
Subject: [PATCH] regulator: stpmic1: support dash in node names

Modify node names to support dash and stay compatible
with older device trees.
pwr_sw1 can now be named pwr-sw1.

Change-Id: I1dcb5d3cf4ce73d50b2f86720ca6ababc943234a
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/291500
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 drivers/regulator/stpmic1_regulator.c | 56 +++++++++++++++------------
 1 file changed, 32 insertions(+), 24 deletions(-)

--- a/drivers/regulator/stpmic1_regulator.c
+++ b/drivers/regulator/stpmic1_regulator.c
@@ -621,26 +621,29 @@ static irqreturn_t stpmic1_boost_irq_han
 }
 
 #define MATCH(_name, _id) \
-	[STPMIC1_##_id] = { \
-		.name = #_name, \
+	{ \
+		.name = _name, \
 		.desc = &stpmic1_regulator_cfgs[STPMIC1_##_id].desc, \
 	}
 
 static struct of_regulator_match stpmic1_matches[] = {
-	MATCH(buck1, BUCK1),
-	MATCH(buck2, BUCK2),
-	MATCH(buck3, BUCK3),
-	MATCH(buck4, BUCK4),
-	MATCH(ldo1, LDO1),
-	MATCH(ldo2, LDO2),
-	MATCH(ldo3, LDO3),
-	MATCH(ldo4, LDO4),
-	MATCH(ldo5, LDO5),
-	MATCH(ldo6, LDO6),
-	MATCH(vref_ddr, VREF_DDR),
-	MATCH(boost, BOOST),
-	MATCH(pwr_sw1, VBUS_OTG),
-	MATCH(pwr_sw2, SW_OUT),
+	MATCH("buck1", BUCK1),
+	MATCH("buck2", BUCK2),
+	MATCH("buck3", BUCK3),
+	MATCH("buck4", BUCK4),
+	MATCH("ldo1", LDO1),
+	MATCH("ldo2", LDO2),
+	MATCH("ldo3", LDO3),
+	MATCH("ldo4", LDO4),
+	MATCH("ldo5", LDO5),
+	MATCH("ldo6", LDO6),
+	MATCH("vref_ddr", VREF_DDR),
+	MATCH("vref-ddr", VREF_DDR),
+	MATCH("boost", BOOST),
+	MATCH("pwr_sw1", VBUS_OTG),
+	MATCH("pwr-sw1", VBUS_OTG),
+	MATCH("pwr_sw2", SW_OUT),
+	MATCH("pwr-sw2", SW_OUT),
 };
 
 static struct regulator_dev *
@@ -757,22 +760,27 @@ static int stpmic1_regulator_probe(struc
 	if (!usb_data)
 		return -ENOMEM;
 
-	for (i = 0; i < ARRAY_SIZE(stpmic1_regulator_cfgs); i++) {
-		if (i == STPMIC1_BOOST) {
+	for (i = 0; i < ARRAY_SIZE(stpmic1_matches); i++) {
+		int id = stpmic1_matches[i].desc->id;
+
+		if (!stpmic1_matches[i].init_data)
+			continue;
+
+		if (id == STPMIC1_BOOST) {
 			rdev =
-			stpmic1_boost_register(pdev, i, &stpmic1_matches[i],
-					       &stpmic1_regulator_cfgs[i],
+			stpmic1_boost_register(pdev, id, &stpmic1_matches[i],
+					       &stpmic1_regulator_cfgs[id],
 					       usb_data);
 
 			usb_data->boost_rdev = rdev;
 		} else {
 			rdev =
-			stpmic1_regulator_register(pdev, i, &stpmic1_matches[i],
-						   &stpmic1_regulator_cfgs[i]);
+			stpmic1_regulator_register(pdev, id, &stpmic1_matches[i],
+						   &stpmic1_regulator_cfgs[id]);
 
-			if (i == STPMIC1_VBUS_OTG)
+			if (id == STPMIC1_VBUS_OTG)
 				usb_data->vbus_otg_rdev = rdev;
-			else if (i == STPMIC1_SW_OUT)
+			else if (id == STPMIC1_SW_OUT)
 				usb_data->sw_out_rdev = rdev;
 		}
 		if (IS_ERR(rdev))
